cmd/mdatagen/loader.go

```diff
@@ -221,13 +221,80 @@ type tests struct {
 	ExpectConsumerError bool   `mapstructure:"expect_consumer_error"`
 }
 
+type counter struct {
+	ValueType   string `mapstructure:"value_type"`
+	Monotonic   bool   `mapstructure:"monotonic"`
+	Synchronous bool   `mapstructure:"synchronous"`
+}
+
+type histogram struct {
+	ValueType string `mapstructure:"value_type"`
+}
+
+func (m *internalMetric) Unmarshal(parser *confmap.Conf) error {
+	if !parser.IsSet("enabled") {
+		return errors.New("missing required field: `enabled`")
+	}
+	return parser.Unmarshal(m)
+}
+
+func (m internalMetric) Type() string {
+	typeStr := ""
+	if m.Counter != nil {
+		switch m.Counter.ValueType {
+		case "int":
+			typeStr += "Int64"
+		case "double":
+			typeStr += "Float64"
+		}
+
+		if !m.Counter.Monotonic {
+			typeStr += "UpDown"
+		}
+		typeStr += "Counter"
+		return typeStr
+	}
+	if m.Histogram != nil {
+		switch m.Histogram.ValueType {
+		case "int":
+			return "Int64Histogram"
+		case "double":
+			return "Float64Histogram"
+		}
+	}
+	return typeStr
+}
+
+type internalMetric struct {
```

Comment:
Can we reuse the existing `metric` struct instead? It'd be nice to provide a consistent interface for metadata.yaml, and potentially config. We would need to add `histogram` there. I don't see a reason to use different naming, e.g., counter vs. sum, given that the end-user outcome is the same.

Reply:
@dmitryax i was trying to tie this to the type of instrument that would be generated as i thought it would be easier to generate the otel instrument, and leave it up to components to use the otel API directly for recording measurements. i can change it though

Reply:
updated the code to remove the need for new internalMetric type. PTAL

Reply:
Thanks. LGTM